src/__tests__/documenter.test.ts

```diff
@@ -1,12 +1,37 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+import fs from "fs";
+import path from "node:path";
+import os from "os";
 import { expect, test } from "vitest";
 
+import { writeComponentsDocumentation } from "@cloudscape-design/documenter";
+
 import componentDefinitions from "../../lib/components/internal/api-docs/components";
 import { getAllComponents } from "./utils";
 
 test.each<string>(getAllComponents())(`definition for %s matches the snapshot`, (componentName: string) => {
   const definition = componentDefinitions[componentName];
   expect(definition).toMatchSnapshot(componentName);
 });
+
+test("internal core API matches snapshot", () => {
+  const tmpDir = path.join(os.tmpdir(), `core-chart-docs-${Date.now()}`);
+  fs.mkdirSync(tmpDir, { recursive: true });
+  try {
```

Comment:
Why is this code within a try block? Will the test still fail if the assertions fail?

Reply:
A possible future improvement is possible, but it requires changes to the documenter. For components, it only exposes the `writeComponentsDocumentation` function, which writes definitions to a file. We can alternatively expose the `documentComponents` function (similar to already exposed `documentTestUtils`). In that case, the temp files will need to be created.